### PR TITLE
Scaladoc on AnyRef#synchronized

### PR DESCRIPTION
**Pull request by eed3si9n**
Ref https://github.com/scala/bug/issues/11310
Ref https://github.com/scala/bug/issues/4086

This PR attempts to surface the Scaladoc of `AnyRef#synchronized`. A quick test can be done using https://github.com/scala/scala/blob/0ff65a000d60b4158776687c7f73940d9dcdf85b/test/scaladoc/run/anyref-doc.scala

```
> partest --srcpath scaladoc --grep anyref-doc
```

As a background, AnyRef is a fictional class for documentation only. This is achieved by a special mode in scaladoc called `-doc-no-compile`, which accepts `src/library-aux`.

### CODE REVIEW

File: src/reflect/scala/reflect/internal/Definitions.scala [hunk 1]
```diff
@@ -1216,6 +1216,8 @@ trait Definitions extends api.StandardDefinitions {
     lazy val Object_isInstanceOf = newT1NilaryMethod(ObjectClass, nme.isInstanceOf_Ob, FINAL | SYNTHETIC | ARTIFACT)(_ => BooleanTpe)
     lazy val Object_asInstanceOf = newT1NilaryMethod(ObjectClass, nme.asInstanceOf_Ob, FINAL | SYNTHETIC | ARTIFACT)(_.typeConstructor)
     lazy val Object_synchronized = enterNewT1Method(ObjectClass, nme.synchronized_, FINAL)(_.typeConstructor)
+    // used to generate scaladoc from AnyRef.scala
+    lazy val AnyRef_synchronized = enterNewT1Method(AnyRefClass, nme.synchronized_, FINAL)(_.typeConstructor)
```
Comment (lrytz) [inline src/reflect/scala/reflect/internal/Definitions.scala]: This looks rather strange, `AnyRef` is defined as a type alias, not sure what entering a method means. In any case, looking up `synchronized` already gives the `Object_synchronized` symbol

```
scala> AnyRefClass.tpe.member(TermName("synchronized")) eq Object_synchronized
val res4: Boolean = true
```

Comment (eed3si9n) [reply src/reflect/scala/reflect/internal/Definitions.scala]: Digging in a bit more deeper, I found 4449c5af3 from 2011. When I comment that out, `synchronized` seems to grab the Scaladoc, so I added 1335969 to save the modified `Object#synchronized` symbol as a `val` so I can convert it back to the original `Object_synchronized`.

File: src/scaladoc/scala/tools/nsc/doc/model/ModelFactory.scala [hunk 2]
```diff
@@ -70,6 +70,17 @@ class ModelFactory(val global: Global, val settings: doc.Settings) {
     case _ => false
   }
 
+  // This unsightly hack closes issue #4086.
```
Comment (lrytz) [inline src/scaladoc/scala/tools/nsc/doc/model/ModelFactory.scala]:
```suggestion
  // This unsightly hack closes issue scala/bug#4086.
```

**Review by lrytz — APPROVED** [review]
LGTM, squash and 🚢

**Comment by eed3si9n** [comment]
Squashed.

**Pull request merged**